confgenerator/config.go

```diff
@@ -25,6 +25,16 @@ import (
 	yaml "gopkg.in/yaml.v2"
 )
 
+// TODO(lingshi): Figure out a cleaner way to do "required" validation.
+// The "omitempty" annotation is reserved to make YAML marshal/unmarshal results reasonable.
```

Comment:
Originally we were relying on the `omitempty` for this. But that makes make YAML marshal/unmarshal results look odd. We might need to find another way for validating "required" field. Annotation based is better than keeping a list. Will revisit this post GA.

Reply:
You can always add your own tag to indicate required...

Reply:
Yes, that's part of the TODO.